crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -115,8 +118,22 @@ impl<'a> DataRef<'a> {
     }
 }
 
+// Type aliases for lock gaurds
+type SharedWriteGuard<T> = ArcRwLockWriteGuard<RawRwLock, T>;
```

Comment:
I'll add that one reason I'm a fan of this is that it makes it much easier to change the concrete lock type in the future.

Reply:
Agree!